### PR TITLE
Drop duplicated dependencies from `pom.xml`

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR addresses [#362](https://github.com/SpineEventEngine/config/issues/362).

In particular, it does the following:

1. Deletes manual applying of `pmd` plugin, which is now applied via our `pmd-settings` script. It eliminates duplicated dependencies on `pmd`.
2. Pulls the latest `config` which contains a [fix](https://github.com/SpineEventEngine/config/pull/368) for the issue. It hides duplicated dependencies on `protoc`, emitting a warning.
3. Enables publication of `dokkaJar` artifact.

Additionally, `J2ObjC.lib` was replaced with `J2ObjC.annotations`, as the former is deprecated:

```kotlin
    const val annotations = "com.google.j2objc:j2objc-annotations:${version}"
    @Deprecated("Please use `annotations` instead.", ReplaceWith("annotations"))
    const val lib = annotations
```

Please note, when pulling the latest `config`, the versions of some libraries have been preserved as they were:

1. ErrorProne kept `2.10.0` due to #181.
2. CommonsCli kept `1.4.0` due to #181.
4. Protobuf kept `3.19.4` due to #183.

The library version is bumped to `2.0.0-SNAPSHOT.74`.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol Please take a look.

**Review by armiol — CHANGES_REQUESTED** [review]
Please take care of the memory consumption. Also, we need to check if the published documentation is fine. You may publish this locally to inspect the artifacts.

File: pom.xml [hunk 1]
```diff
@@ -254,20 +249,50 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-core</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>gfm-plugin</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:267]: @sonatype-lift ignore

File: pom.xml [hunk 2]
```diff
@@ -254,20 +249,50 @@ all modules and does not describe the project structure per-subproject.
...
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-base</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
+    <groupId>org.jetbrains.dokka</groupId>
+    <artifactId>dokka-core</artifactId>
+    <version>1.6.20</version>
+  </dependency>
+  <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:262]: @sonatype-lift ignore

**Comment by yevhenii-nadtochii** [comment]
@armiol PTAL

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**